### PR TITLE
Update the test models and CI dependencies

### PR DESCRIPTION
**Pull request by raimis**
- [x] Update dependencies
- [x] Update to PyTorch 1.11
  - [x] Regenerate the test models

### CODE REVIEW

**Comment by raimis** [comment]
Uff! Still the dependencies are missing for Python 3.10

**Comment by raimis** [comment]
So, at the moment, it is only possible to update CUDA.

**Comment by raimis** [comment]
- Fixed the tests with PyTorch 1.11. The test models have been generated with some old PyTorch and PyTorch 1.11 broke backward compatibility. This should allow to build the `conda-forge` packages  for PyTorch 1.11 (https://github.com/conda-forge/openmm-torch-feedstock/pull/17).
- After, it will be possible to fix the tutorial test, which are broken because the `conda-forge` migrated to PyTorch 1.11 (https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2812) and OpenMM-Torch was left behind.

File: tests/generate.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import torch as pt
+
+class Central(pt.nn.Module):
+    def forward(self, pos):
+        return pos.pow(2).sum(dim=1).abs().sum()
```
Comment (peastman) [inline tests/generate.py:5]: I think this and the similar ones below can be simplified to just `pos.pow(2).sum()`.  Squaring the elements guarantees all of them are positive, so `abs()` does nothing.  And that means the intermediate sum also is unnecessary

Comment (raimis) [reply tests/generate.py]: Good point.

Comment (raimis) [reply tests/generate.py]: Done!

**Comment by raimis** [comment]
@peastman any more comments?

**Comment by peastman** [comment]
Looks good!  I see there's a test failure.  It looks like the same problem as #76.

**Comment by mikemhenry** [comment]
I'll see if there is a quick re-render and rebuild of nnpops fixes this.

**Pull request merged**